### PR TITLE
Implement Method to Mask Content in Adaptive Authentication Scripts

### PR DESCRIPTION
**Pull request by dhaura**
- **Method Name**: `getMaskedValue`
- **Input Parameters**: `String value` [Value to be masked] (ex: johndoe@gmail.com)
- **Return Type**: `String` [Masked value] (ex: j***************m)
- **Functionality**: 
    - Always mask the given value and return it.

```js
var onLoginRequest = function(context) {
    executeStep(1, {
        onSuccess: function(context) {
                var email = context.currentKnownSubject.username;
                Log.info("Email of the logged user : " + getMaskedValue(email));
            }
        },
    });
};
```

## Automation tests
 - Unit tests 
   > Covered with this PR.
 - Integration tests
   > N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

 

## Refrences
[1] - https://github.com/wso2/carbon-identity-framework/blob/e100a46b3431bbc676c2410ad00adc5926f2c0fb/components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java#L255-L261
[2] - https://github.com/wso2/carbon-identity-framework/blob/e100a46b3431bbc676c2410ad00adc5926f2c0fb/components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementAuditLogger.java#L193
[3] - https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/61d5bbb98d68b031a3540b7fff48e8f090474455/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java#L188
[4] - https://github.com/wso2-extensions/identity-conditional-auth-functions/tree/master/components/org.wso2.carbon.identity.conditional.auth.functions.utils

### CODE REVIEW

File: components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/GetMaskedValueFunctionImpl.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Implementation of  {@link GetMaskedValueFunction}.
+ */
+public class GetMaskedValueFunctionImpl implements GetMaskedValueFunction {
+
+    @Override
+    @HostAccess.Export
+    public String getMaskedValue(String value) {
+
+        return  LoggerUtils.getMaskedContent(value);
```
Comment (UdeshAthukorala) [inline components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/GetMaskedValueFunctionImpl.java:33]: Shall we fix the formatting here. (Additional space)

Comment (dhaura) [reply components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/GetMaskedValueFunctionImpl.java]: Addressed with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/167/commits/4b01845017ca7ef6ca737f08bfdbda6d95f466fe

**Review by UdeshAthukorala — APPROVED** [review]

File: components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/internal/UtilsFunctionServiceComponent.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ * Copyright (c) 2023-2024, WSO2 LLC. (http://www.wso2.com).
```
Comment (sumedhe) [inline components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/internal/UtilsFunctionServiceComponent.java:2]: Are we following year-range format?

Comment (dhaura) [reply components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/internal/UtilsFunctionServiceComponent.java:2]: Yes, this is the currently followed format.
- https://github.com/wso2/identity-apps/pull/3969#discussion_r1591758664

**Pull request merged**